### PR TITLE
More input region tests

### PR DESCRIPTION
**Pull request by wmww**
* `InputRegionPointerEnterTest` -> `SurfaceInputRegion`
* Added `pointer_enter_leave_subsurface`
* Added `touch_inside_outside_subsurface`

### CODE REVIEW

**Comment by wmww** [comment]
Hmm... you're probably right. Will think of something better.

**Comment by wmww** [comment]
improved the names a bit, open to further suggestions.

**Comment by AlanGriffiths** [comment]
> @wmww improved the names a bit, open to further suggestions.

Well the names are longer, but still don't indicate what is being tested.  If you see a failure in "SurfaceInputRegion.touch_inside_and_outside_subsurfaces_input_region" can you guess what it means?

Better test names would be:

    SurfaceInputRegion.a_touch_inside_region_is_seen_by_subsurface
    SurfaceInputRegion.a_touch_outside_input_region_is_not_seen_by_subsurface
    SurfaceInputRegion.a_touch_is_seen_by_expected_subsurface

**Comment by AlanGriffiths** [comment]
Here's why it matters: https://juristr.com/blog/2012/07/testing-and-single-responsibility/

File: tests/surface_input_regions.cpp [hunk 1]
```diff
@@ -180,6 +221,130 @@ TEST_P(InputRegionPointerEnterTest, touch_enter_leave)
...
+                    wl_fixed_from_int(params.initial_y + params.dy))));
+
+    /* move pointer back; it should now be outside the surface */
+    pointer.move_by(-params.dx, -params.dy);
+
+    client.roundtrip();
+    EXPECT_THAT(client.focused_window(), Ne(wl_surface));
+}
+
+TEST_P(SurfaceInputRegion, touch_out_subsurface_input_region_is_seen)
```
Comment (AlanGriffiths) [inline tests/surface_input_regions.cpp]: "out" => "outside"?

**Pull request merged**